src/Explorer/Explorer.tsx

```diff
@@ -212,6 +207,23 @@ export default class Explorer {
     this.refreshExplorer();
   }
 
+  public async initiateNotebooks(): Promise<void> {
```

Comment:
give a better name? there is already a very similar "initializeNotebooks" or something. Looks like you do this twice:

useNotebook.subscribe(
      async () => {
        this.initiateNotebooks();
...        
},

is it intentional? what is it trying to achieve?

Reply:
updated method name